core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/AuditLog.java

```diff
@@ -18,165 +18,156 @@
 
 package org.wso2.carbon.utils;
 
+import org.slf4j.MDC;
+
 import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.wso2.carbon.CarbonConstants.LogEventConstants.CORRELATION_ID_MDC;
 
 /**
  * Audit log.
  */
 public class AuditLog {
 
-    private String logId;
-    private Instant recordedAt;
-    private String clientComponent;
-    private String correlationId;
-    private String initiatorId;
-    private String initiatorName;
-    private String initiatorType;
-    private String eventType;
-    private String targetId;
-    private String targetName;
-    private String targetType;
-    private String dataChange;
-
-    public AuditLog(String logId, Instant recordedAt, String clientComponent, String correlationId,
-                    String initiatorId, String initiatorName, String initiatorType, String eventType,
-                    String targetId, String targetName, String targetType, String dataChange) {
-
-        this.logId = logId;
-        this.recordedAt = recordedAt;
-        this.clientComponent = clientComponent;
-        this.correlationId = correlationId;
-        this.initiatorId = initiatorId;
-        this.initiatorName = initiatorName;
-        this.initiatorType = initiatorType;
-        this.eventType = eventType;
-        this.targetId = targetId;
-        this.targetName = targetName;
-        this.targetType = targetType;
-        this.dataChange = dataChange;
-    }
+    private final String id;
+    private final String recordedAt;
+    private final String requestId;
+    private final String initiatorId;
+    private final String initiatorType;
+    private final String targetId;
+    private final String targetType;
+    private final String action;
+    private final Map<String, Object> data;
 
-    public AuditLog() {
+    private AuditLog(AuditLogBuilder auditLogBuilder) {
 
+        this.id = auditLogBuilder.id;
+        this.recordedAt = auditLogBuilder.recordedAt;
+        this.requestId = auditLogBuilder.requestId;
+        this.initiatorId = auditLogBuilder.initiatorId;
+        this.initiatorType = auditLogBuilder.initiatorType;
+        this.targetId = auditLogBuilder.targetId;
+        this.targetType = auditLogBuilder.targetType;
+        this.action = auditLogBuilder.action;
+        this.data = auditLogBuilder.data;
     }
 
-    public String getLogId() {
+    public String getId() {
 
-        return logId;
+        return id;
     }
 
-    public void setLogId(String logId) {
-
-        this.logId = logId;
-    }
-
-    public Instant getRecordedAt() {
+    public String getRecordedAt() {
```

Comment:
changed to String. Earlier, the audit logs had complex object as recordedAt. 
```
"recordedAt":{"seconds":1688717227,"nanos":561705000
```
When publishing the audit logs, google gson library convert the Instant class to a complex attribute.